src/java/org/jivesoftware/openfire/privacy/PrivacyListProvider.java

```diff
@@ -86,11 +97,9 @@ public PrivacyListProvider() {
 
         // Load the total number of privacy lists in the database. We're looking
         // for the (very common) special case that there are no privacy lists stored.
-        // In that case, we can optimize away many database calls. In the future, a
-        // better general-case solution may be to cache all privacy lists defined
-        // if there are less than, say, 500.
-        privacyListCount = new AtomicInteger(0);
-        loadPrivacyListCount();
+        // In that case, we can optimize away many database calls. 
+        privacyListNotUsed = new AtomicBoolean(true);
```

Comment:
Holy double negation, Batman! It's probably easier to name this "databaseContainsPrivacyLists", which is what we check for. Also, the comment above this line refers to a 'number', which is no longer the case.